lib/cadet/assessments/assessments.ex

```diff
@@ -1099,23 +1218,54 @@ defmodule Cadet.Assessments do
     end
   end
 
-  defp insert_or_update_answer(submission = %Submission{}, question = %Question{}, raw_answer) do
+  defp insert_or_update_answer(
+         submission = %Submission{},
+         question = %Question{},
+         raw_answer,
+         user_id
+       ) do
     answer_content = build_answer_content(raw_answer, question.type)
 
-    answer_changeset =
-      %Answer{}
-      |> Answer.changeset(%{
-        answer: answer_content,
-        question_id: question.id,
-        submission_id: submission.id,
-        type: question.type
-      })
+    if question.type == :voting do
+      insert_or_update_voting_answer(user_id, answer_content)
+    else
+      answer_changeset =
+        %Answer{}
+        |> Answer.changeset(%{
+          answer: answer_content,
+          question_id: question.id,
+          submission_id: submission.id,
+          type: question.type
+        })
 
-    Repo.insert(
-      answer_changeset,
-      on_conflict: [set: [answer: get_change(answer_changeset, :answer)]],
-      conflict_target: [:submission_id, :question_id]
-    )
+      Repo.insert(
+        answer_changeset,
+        on_conflict: [set: [answer: get_change(answer_changeset, :answer)]],
+        conflict_target: [:submission_id, :question_id]
+      )
+    end
+  end
+
+  defp insert_or_update_voting_answer(user_id, answer_content) do
+    ans =
+      answer_content
+      |> Enum.map(fn ans ->
+        {submission_id, score} = ans
+
+        SubmissionVotes
+        |> where(submission_id: ^submission_id)
+        |> where(user_id: ^user_id)
+        |> Repo.one()
+        |> SubmissionVotes.changeset(%{score: score})
+        |> Repo.update()
+      end)
+
+    [status | _] = ans
```

Comment:
You are checking only the result of the first update?

Please use a Multi transaction for this